### DISCUSSION TITLE
adb host as parameter

### DISCUSSION THREAD

**Discussion by QAutomatron** [Banno/Gordon#24] [Ideas]
`JadbConnection` supports using an external host for `adb` commands. It would be nice to have an option to set it from Gordon configuration. 

Use cases: access host machine from docker container. Or access devices from another host in network.

**Comment by joshschriever** [comment]
As of #39 we also use bundletool for installs, and I don't think that supports external adb host yet. There's [a PR for it](https://github.com/google/bundletool/pull/93), but that's stale. Hopefully eventually we can support this somehow.

**Comment by sleekweasel** [comment]
An alternative to telling bundletool how to talk to remote hosts that I didn't pursue because I got my patch working: run a proxy forwarding socket, something like 'socat', to give yourself a local port that connects to the remote machine.

**Comment by joshschriever** [comment]
@QAutomatron As of #95 (released in [v1.7.0](https://github.com/Banno/Gordon/releases/tag/1.7.0)), we got rid of jadb and started using bundletool for everything. Bundletool should respect the `ANDROID_ADB_SERVER_PORT` env variable if set, so using that along with something like socat, as @sleekweasel mentioned, could work.
